flambe/compile/downloader.py

```diff
@@ -176,29 +183,33 @@ def download_manager(path: str):
             raise ValueError(f"Path: '{path}' does not exist locally.")
 
     else:
+        trailing_url = url.path[:-1] if url.path.endswith('/') else url.path
+        fname = trailing_url[trailing_url.rfind('/') + 1:]
```

Comment:
Could this result in file collisions? Why not use the key in the resources dict for the file name?

Reply:
Yes, def we can have collisions. Using the key in the resources is a good idea definitely.

Reply:
Fixed.